File: TinyClaw/Views/Chat/SpecialistBadge.swift
import SwiftUI
import TinyClawCore

struct SpecialistBadge: View {
    let intent: Intent

    private var color: Color {
        switch intent {
        case .chat: return .green
        case .summarize: return .red
        case .task: return .blue
        case .event: return .purple
        }
    }

    var body: some View {
        Text(intent.rawValue.uppercased())
            .font(.system(size: 9, weight: .semibold))
            .foregroundStyle(color)
            .padding(.horizontal, 8)
            .padding(.vertical, 2)
            .background(color.opacity(0.15))
            .clipShape(RoundedRectangle(cornerRadius: 4))
    }
}

File: lib/views/chat/specialist_badge.dart
import 'package:flutter/material.dart' hide Intent;
import '../../core/intent.dart';

class SpecialistBadge extends StatelessWidget {
  final Intent? intent;

  const SpecialistBadge({super.key, this.intent});

  Color _badgeColor() {
    return switch (intent) {
      Intent.chat => Colors.grey,
      Intent.summarize => Colors.orange,
      Intent.task => Colors.blue,
      Intent.event => Colors.purple,
      Intent.note => Colors.amber,
      Intent.habit => Colors.green,
      Intent.draft => Colors.indigo,
      Intent.search => Colors.pink,
      null => Colors.grey,
    };
  }

  String _label() {
    return intent?.name ?? 'unknown';
  }

  @override
  Widget build(BuildContext context) {
    return Container(
      padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 2),
      decoration: BoxDecoration(
        color: _badgeColor().withValues(alpha: 0.15),
        borderRadius: BorderRadius.circular(8),
      ),
      child: Text(
        _label(),
        style: TextStyle(
          fontSize: 10,
          fontWeight: FontWeight.w600,
          color: _badgeColor(),
        ),
      ),
    );
  }
}
